### DISCUSSION TITLE
How to preserve ampersand character

### DISCUSSION THREAD

**Discussion by smackesey** [remarkjs/remark#951] [Q&A] [answered]
I can't figure out how to preserve the ampersand character in a markdown string:

```
const remark = require('remark');
const Remark = remark();

// Every one of the below adds a backslash before &, i.e. output is "abc\&def":

Remark.processSync('a&b').toString();
Remark.processSync('a\&b').toString();
Remark.processSync('a\\&b').toString();
Remark.processSync('a&amp;b').toString();

// However, if the & is standalone, this doesn't happen:
Remark.processSync('a & b').toString()  // "a & b"
```

How can I prevent `Remark` from escaping ampersands that occur inside of a word?

**Comment by wooorm** [comment, marked as answer]
You can’t. You don’t need to.

**Reply by smackesey** [reply]
Well good to know that it's not possible, but I _do_ need to. We use `Remark` to manage user-provided Markdown, and in some places we convert to plaintext with `https://www.npmjs.com/package/remark-plain-text`. Unfortunately the plain-text conversion introduces extraneous "\" if the user included "&" as shown above. I thought it might be the plaintext plugin but it's the base remark parser/serializer.

Guess I need to find another way to convert to plaintext, or maybe just manually remove them.

**Reply by ChristianMurphy** [reply]
> Unfortunately the plain-text conversion introduces extraneous "\\" if the user included "&" as shown above. 

The escape doesn't change how the markdown is rendered.
remark (and all of unified) works with abstract syntax trees, the raw text may not be identical before and after running a transform, but how it is interpreted/displayed stays consistent.

**Reply by wooorm** [reply]
remark is about markdown, which is not the same as plain text. What “plain text” is, is super vague though.

See https://github.com/remarkjs/.github/blob/main/support.md, perhaps you can reframe your question.
